### ISSUE TITLE
Allow setting `use_param_file` and `set_param_file_format` on frozen `action.args()` objects, or allow creating mutable copies of `args` objects

### ISSUE THREAD

**Issue by brentleyjones** [bazelbuild/bazel#20746]
### Description of the feature request:

When using `Target.actions.args` the returned `args` objects are frozen, and thus `use_param_file` and `set_param_file_format` can't be called on them. There should be a way to change the params file options for those objects, or better yet a way to make a mutable copy that we can set them on.

### What underlying problem are you trying to solve with this feature?

Some actions (e.g. `CppCompileAction`) return `args` objects without params file options set on them, even if the underlying action did have those options set on them. While for this exact action we can possibly use the experimental and undocumented `--experimental_use_cpp_compile_action_args_params_file` flag, other actions might not have set params file options. This is an issue when a shadowed action, or a new action created in an aspect, uses these args objects, since the command-line length can exceed OS limits.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 7.0.0

**Comment by comius** [comment]
I'm marking P4, becuase I think, this will resolve when we further Starlarkfiy C++ rules.

I have an example of CppLinkAction, which will after my last PR, have the params file correctly set and can be reused.

If I correctly understand the description, there would be no need to set params file if it was already set correctly, right?

**Comment by brentleyjones** [comment]
Correct. But some custom rules might not set the params file because they are small enough to not need it, but when those args are used with other args in a new action in an aspect/extension/etc., they could exceed the OS limits. So I still see a need for this in the general case.

**Comment by brentleyjones** [comment]
Another example this needs to be more than a P4: https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2878
